x-pack/plugins/stack_connectors/server/connector_types/email/send_email.ts

```diff
@@ -70,13 +71,14 @@ export async function sendEmail(
   connectorTokenClient: ConnectorTokenClientContract
 ): Promise<unknown> {
   const { transport, content } = options;
-  const { message } = content;
-  const messageHTML = htmlFromMarkdown(logger, message);
+  const { message, messageHTML } = content;
+
+  const renderedMessage = messageHTML ?? htmlFromMarkdown(logger, message);
```

Comment:
hmm I wonder if we should instead pass a boolean that determines whether this code calls `htmlFromMarkdown`. What do you think @pmuellr ?

That way we can just leverage the `message` instead of adding a new one specific to HTML.

I think I'm envisioning the schema looking like this:

```
bcc: schema.arrayOf(schema.string(), { defaultValue: [] }),
  subject: schema.string(),
  message: schema.string(),
  isHTML: schema.nullable/maybe(schema.boolean()),
```

Reply:
hmmm ... good question.  It is a bit weird having two of these.  Another option would to turn this into a subcommand, one for markdown, one for html - but actually not sure we can or would want to "migrate" existing connectors if they had to change.

Can't think off the top of my head any other "dialects" we'd want to support, if we'd want to with an even more general approach like

```js
language: 'markdown' | 'html' | undefined
```

So, seems like either way is fine - guessing Jonathan's approach would probably result in less code, in the end.

Reply:
Also thinking about https://github.com/elastic/kibana/issues/94963 and https://github.com/elastic/kibana/issues/94678 - the parts about the markdown-it `html: true`.

We definitely want to implement that, it's not quite clear if we need some global config to enable using it at all (in case some org thinks it's too unsafe to allow), and then presumably it would be similar to `isHtml`, except I guess I'd call it `allowHtml`.

Could you check running your template through the markdown processor with that flag set, see if it hopefully does nothing? :-). 

If so, I think one more step of having some global config to enable it's use, would be good.  Actually, in terms of customers who may consider this "too unsafe to allow", maybe we should do this anyway - honestly not sure of how big of a concern this is.  Prolly check with security?  Would need to be allow-listed in cloud, threaded through our config utilities, etc.

Ideally we could make this a UX toggle, but I'd be quite happy to defer that - and there are likely some IaC customers (for example, using Ansible) who could use it without the UX.

Reply:
> hmm I wonder if we should instead pass a boolean that determines whether this code calls `htmlFromMarkdown`. What do you think @pmuellr ?
> 
> That way we can just leverage the `message` instead of adding a new one specific to HTML.
> 
> I think I'm envisioning the schema looking like this:
> 
> ```
> bcc: schema.arrayOf(schema.string(), { defaultValue: [] }),
>   subject: schema.string(),
>   message: schema.string(),
>   isHTML: schema.nullable/maybe(schema.boolean()),
> ```

Currently we are using both params `message` for plain text message and `messageHTML` with HTML string.
If we use boolean flag, we will still have to add logic here to keep both type of messages, right?

Reply:
Hey @jonathan-buttner and @pmuellr! The reason we want a second variable (`messageHTML`) instead of a boolean is that we want to provide both an HTML message and a plaintext message on the same email. We followed the schema enforced by `nodemailer` which accepts a `message (text)` and an `messageHTML (html)`. Any ideas/alternatives? Stripping out the HTML ourselves seems an overkill.

> Another option would to turn this into a subcommand

This would be ideal but as you said migrating existing connectors would be hard and seems to be out of the scope of this PR.

Reply:
> Also thinking about #94963 and #94678 - the parts about the markdown-it `html: true`.
> 
> We definitely want to implement that, it's not quite clear if we need some global config to enable using it at all (in case some org thinks it's too unsafe to allow), and then presumably it would be similar to `isHtml`, except I guess I'd call it `allowHtml`.
> 
> Could you check running your template through the markdown processor with that flag set, see if it hopefully does nothing? :-).
> 

I tested with `html:true` option and it looks like it:
<img width="1214" alt="image" src="https://github.com/elastic/kibana/assets/117571355/e3a11954-dd19-4139-a58b-acd274459857">

Reply:
I misunderstood what was happening here. Sorry for the confusion. Since it looks like passing html to markdown-it isn't working and we need both `message` and `messageHTML` to be separate entities I think what you have is good.

Reply:
> we want to provide both an HTML message and a plaintext message on the same email.

Ah ... ya, I was thinking about that last night - if we went my approach, then the "plain text" version would be full of icky HTML.

So the way this stands now, a customer could make use of the `messageHTML` property in infrastructure-as-code built rules themselves, but still no UX for it.  Not that we want to advertise it.  Will need to think a bit more how this might play with the eventual "allow HTML in your markdown" enhancement, but doesn't feel like there would be any issues ...

Reply:
Thanks, both of you!